### ISSUE TITLE
[1.5.0] Reset password link and overlays broken

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#2605]
This seems to be related to the move to the RegisterComponent API and is possibly due to the fact that much of this happens in overlays, but I'm not fully convinced it may just be a rogue component not being registered correctly.

To Reproduce:
1. checkout `development` or `marketplace`
2. Click "reset password"
3. Observer the following error
```
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in. Check the render method of `Forgot`.
    in Forgot (created by Reaction(ForgotPassword))
    in Reaction(ForgotPassword) (created by Reaction(Login))
    in Reaction(Login)
printWarning @ modules.js:84625
modules.js:84792 Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in. Check the render method of `Forgot`.
    at invariant (modules.js:84785)
    at instantiateReactComponent (modules.js:106368)
    at instantiateChild (modules.js:106176)
    at modules.js:106203
    at traverseAllChildrenImpl (modules.js:107832)
    at traverseAllChildren (modules.js:107927)
    at Object.instantiateChildren (modules.js:106202)
    at ReactDOMComponent._reconcilerInstantiateChildren (modules.js:105754)
    at ReactDOMComponent.mountChildren (modules.js:105793)
    at ReactDOMComponent._createInitialChildren (modules.js:103004)
```

Also happens and I first noticed it when trying to invite a user from email, the email link does not open the overlay.

I've spent a couple hours trying to debug and am out of ideas, but it's a critical blocker for inviting new shop owners on marketplace as well as for any other user being able to reset their password.

**Comment by jshimko** [comment]
I'm 99.9% positive I know what that is.  I'm traveling today, but I'll knock this out at some point this weekend.

**Comment by spencern** [comment]
Thanks Jeremy. We'll hold off on finishing up #2538 until that.

**Comment by spencern** [comment]
@jshimko Assigning you on this one, but if you can't figure it out quickly let us know what to look for and we can pop someone else on this.

**Comment by spencern** [comment]
Fixed by #2612
